src/db.rs

```diff
@@ -287,17 +296,21 @@ impl KbsDb {
         Ok(())
     }
 
-    pub async fn get_keyset_policy(&self, keysetid: &str) -> Result<policy::Policy> {
+    pub async fn get_keyset_policy(&self, keysetid: &str) -> Result<Option<policy::Policy>> {
         let query_str = "SELECT polid FROM keysets WHERE keysetid = ?";
         let new_query_str = self.replace_binds(query_str);
-
         let pol_row = sqlx::query(&new_query_str)
             .bind(keysetid)
             .fetch_one(&self.dbpool)
             .await?;
-        let pol = pol_row.try_get::<i64, _>(0)? as u64;
-        let secret_policy = self.get_policy(pol).await?;
-        Ok(secret_policy)
+
+        match pol_row.try_get::<i64, _>(0) {
+            Ok(pid) => {
+                let secret_policy = self.get_policy(pid as u64).await?;
+                Ok(Some(secret_policy))
```

Comment:
As @fitzthum said above: Can probably combine these lines.

Reply:
Done

Reply:
Looks like this was changed in the above function but not this one.